### DISCUSSION TITLE
VLC mpv huge package size

### DISCUSSION THREAD

**Discussion by goyalyashpal** [msys2/MSYS2-packages#3587] [General]
Hi!

- The video player _packages_ are huge in sizes of both download and install
- ... 10-20X the size of their corresponding official releases

|  | packages (MiB) | official releases (MiB) |
| -- | -- | -- |
mpv | [mingw-w64-mpv] : <ul><li>390-410 MiB \| 2290-2570 MiB</li></ul> | [mpv] : 20-30 MiB \| 100 MiB
vlc | [mingw-w64-vlc] : <ul><li>420-440 MiB \| 2430-2730 MiB</li></ul> | [vlc] : 40-70 MiB \| 200 MiB

### Questions:
- What's the reason 🤔 
- Is there a general metric where i should expect the packages to be _larger_ than releases,
- ... and where i should expect packages to be _smaller_ than releases

[mingw-w64-vlc]: https://packages.msys2.org/base/mingw-w64-vlc
[mingw-w64-mpv]: https://packages.msys2.org/base/mingw-w64-mpv
[vlc]: https://www.videolan.org/vlc/download-windows.html
[mpv]: https://sourceforge.net/projects/mpv-player-windows/files/

<details>

```console
$  pacboy debug -S vlc:c <<< n
:: Executing pacman --color auto -S mingw-w64-clang-x86_64-vlc...

Packages (206)

Total Download Size:    422.34 MiB
Total Installed Size:  2435.02 MiB

$  pacboy debug -S vlc:x <<< n
:: Executing pacman --color auto -S mingw-w64-x86_64-vlc...

Packages (202)

Total Download Size:    435.54 MiB
Total Installed Size:  2732.94 MiB

$  pacboy debug -S vlc:u <<< n
:: Executing pacman --color auto -S mingw-w64-ucrt-x86_64-vlc...

Packages (206)

Total Download Size:    438.27 MiB
Total Installed Size:  2737.95 MiB
```

</details>

**Comment by Biswa96** [comment]
In the pacboy command, you are calculating the size of vlc and all dependencies.

**Reply by goyalyashpal** [reply]
but the release files too contain all that is required to install and use those

**Reply by Biswa96** [reply]
The mingw packages contain static and shared libraries and header files also.

**Reply by goyalyashpal** [reply]
can u elaborate a bit?

**Reply by Biswa96** [reply]
See the file list of any dependencies here https://packages.msys2.org/package/mingw-w64-x86_64-vlc?repo=mingw64

**Reply by goyalyashpal** [reply]
ohkay.

OT: is the situation similar on linux distros?

**Reply by Biswa96** [reply]
Depends on the Linux distributions.

**Comment by goyalyashpal** [comment]
> * Is there a general metric where i should expect the packages to be _larger_ than releases,
> * ... and where i should expect packages to be _smaller_ than releases

any input regarding this?

**Reply by Biswa96** [reply]
more code + more dependencies = bigger package

**Comment by goyalyashpal** [comment]
just one example of the otherwise obvious thing: it's not always that one repo will have smaller package sizes. _and_ `:x` i.e. more softwares are availeble in normal mingw's repo, latest example being: octave `8.1.0-1` which's not available for clang repo

for package <tt>[meld3]</tt> the "Visual [diff] and merge tool". here too, the v3.22 msi installer from [meldmerge.org] is just 36 MiB

[meldmerge.org]: https://meldmerge.org
[meld3]: http://packages.msys2.org/base/mingw-w64-meld3
[diff]: https://unix.stackexchange.com/questions/670458/670486#670486 "using-kate-text-editor-how-to-compare-two-files-and-find-differences"

<details open>

```console
$  pacboy debug -S meld3:x <<< n
Packages (63)
Total Download Size:    87.16 MiB
Total Installed Size:  578.62 MiB

$ pacboy debug -Ss meld3:c <<< n
Packages (67)
Total Download Size:    92.17 MiB
Total Installed Size:  583.06 MiB

$  pacboy debug -S meld3:u <<< n
Packages (67)
Total Download Size:    89.79 MiB
Total Installed Size:  583.73 MiB
```

**2023.03.06:**
```console
$ pacboy -S cppcheck:x <<< n
Packages (4)
Total Download Size:    4.05 MiB
Total Installed Size:  19.61 MiB

$ pacboy -S cppcheck:c <<< n
Packages (7)
Total Download Size:    5.33 MiB
Total Installed Size:  32.23 MiB

$ pacboy -S cppcheck:u <<< n
Packages (7)
Total Download Size:    5.01 MiB
Total Installed Size:  22.76 MiB
```

2023.03.11

```console
$ pacboy -S gnome-calculator:x <<< n 2>/dev/null | grep -Eo "Packages \(\.*\)|.*Size:.*"
Packages (90)
Total Download Size:   117.60 MiB
Total Installed Size:  786.64 MiB

$ pacboy -S gnome-calculator:c <<< n 2>/dev/null | grep -Eo "Packages \(\.*\)|.*Size:.*"
Packages (95)
Total Download Size:   122.94 MiB
Total Installed Size:  781.45 MiB

$ pacboy -S gnome-calculator:u <<< n 2>/dev/null | grep -Eo "Packages \(\.*\)|.*Size:.*"
Packages (94)
Total Download Size:   120.23 MiB
Total Installed Size:  792.15 MiB
```

</details>
